src/bootstrap/src/utils/shared_helpers.rs

```diff
@@ -46,7 +46,7 @@ pub fn dylib_path() -> Vec<std::path::PathBuf> {
 /// Given an executable called `name`, return the filename for the
 /// executable for a particular target.
 pub fn exe(name: &str, target: &str) -> String {
-    if target.contains("windows") || target.contains("cygwin") {
+    if target.contains("windows") {
```

Comment:
I'm confused by this.  I just had to re-discover this change because bootstrap panicked not finding `rustc-main` when it was named `rustc-main.exe`, cross-compiling from Windows to Cygwin.  While Cygwin does have 'magic' that will let you access a file X.exe by the name X, some other host wouldn't know to do that.  What was the problem that reverting this change addressed, and is there maybe some other way to fix it?

Reply:
The rustc wrapper of bootstrap failed to determine the correct parameters. The wrapper is executed like `/foo/rustc /bar/rustc --baz`.

I think there should be other solutions, but it's only a bootstrap tool. So rely on the cygwin magic is fine.

Reply:
I thought that bootstrap tool was the general way to cross-compile rust (build != host), and that should work for build != cygwin host = cygwin

Reply:
Well it might be complex then... Although the bootstrap tool *should* provide a way to cross-compile, it's not that common to cross compile from Windows to a Unix platform (Cygwin). Yet it would be better if it can support that case.

Reply:
Linux to Cygwin should work though?  I might try that and see.

Reply:
OK, I see what the problem is - `/proc/self/exe`'s target is without the `.exe` extension, so the comparison in bootstrap's rustc wrapper would not match on a Cygwin host if this function returned the name with `.exe` appended.  However, on a non-Cygwin host, bootstrap needs to know that executables for a Cygwin target will have a `.exe` extension.  I'm not sure how to reconcile these two different uses of `exe` function.

Reply:
> Linux to Cygwin should work though?

The great pioneer @Ookiineko succeeded compiling rustc from Fedora to Cygwin.

Reply:
I would bet that would require this `exe` function to append `.exe` though

Reply:
Actually yes. They appended `.exe`. I removed it.

Reply:
Maybe there should be a special case for Cygwin host? Otherwise, `.exe` would be expected.

Reply:
I think there has to be a special case for Cygwin host somewhere, because `current_exe` returns without the `.exe` extension on a Cygwin host, but the real filename does have a `.exe` extension (which other hosts would have to know).